### PR TITLE
server: use axum

### PR DESCRIPTION
**Pull request by hi-rustin**
* use axum
* better jwt
* use this error
* support cors

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -6,17 +6,16 @@ edition = "2021"
...
 serde_json = "1.0"
-base64 = "0.13.0"
-url = { version = "2", features = ["serde"] }
-percent-encoding = "2.1.0"
 rust-crypto = "^0.2"
+headers = "0.3"
 jsonwebtoken = "7"
-rand = "0.8.4"
+once_cell = "1.8"
+thiserror = "1"
```
Comment (gozssky) [inline Cargo.toml:21]: Do we need newline at end of file?

File: src/auth.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    let mut verified = false;
+    storage
+        .read_only(|state| {
+            verified = state.users.iter().any(|u| {
+                u.username == req.username
+                    && pbkdf2_check(&req.password, &u.password_hash).ok().unwrap()
+            });
+        })
+        .await;
+    if true {
```
Comment (gozssky) [inline src/auth.rs]:
```suggestion
    if verified {
```

**Review by gozssky — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (hi-rustin) [reply Cargo.toml]: Added.

File: src/model.rs [hunk 3]
```diff
@@ -1,65 +1,74 @@
...
 }
 
-#[derive(Debug, Clone, Default, Serialize, Deserialize)]
-#[serde(default)]
-pub struct ErrorResponse {
-    pub error: String,
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct State {
+    pub users: Vec<User>,
+    pub secret: String,
```
Comment (gozssky) [inline src/model.rs:67]:
```suggestion
```

**Pull request merged**